mono/utils/mono-threads.c

```diff
@@ -1775,6 +1775,39 @@ mono_threads_close_thread_handle (MonoThreadHandle *thread_handle)
 	mono_refcount_dec (thread_handle);
 }
 
+/*
+ * mono_threads_open_native_thread_handle:
+ *
+ *  Duplicate the handle. The handle needs to be closed by calling
+ *  mono_threads_close_native_thread_handle () when it is no longer needed.
+ */
+MonoNativeThreadHandle
+mono_threads_open_native_thread_handle (MonoNativeThreadHandle thread_handle)
+{
+#ifdef HOST_WIN32
+	BOOL success = FALSE;
+	HANDLE new_thread_handle = NULL;
+
+	g_assert (thread_handle && thread_handle != INVALID_HANDLE_VALUE);
+	success = DuplicateHandle (GetCurrentProcess (), thread_handle, GetCurrentProcess (), &new_thread_handle, 0, FALSE, DUPLICATE_SAME_ACCESS);
+	g_assertf (success, "Failed to duplicate thread handle");
```

Comment:
This can fail under low memory and imho should be removed.

Reply:
`return success ? new_thread_handle : NULL;`

Reply:
And then really `return DuplicateHandle (…) ? new_thread_handle : NULL;`

Reply:
This code was mostly moved from old implementation, but I can make it more compact.

Reply:
The assert has been there for a long time, so lets keep it for now.

Reply:
I think it is all ok, just subtle in the NULL vs. INVALID_HANDLE_VALUE handling, except that low memory should not lead to assertion failures.

Reply:
Checked history of the assert and it looks "quite" recently added, 2883e562e2d2, so agree that we should get rid of it.